### PR TITLE
feat: fields, schemas, and sources binary output for the textplan parser

### PR DESCRIPTION
**Pull request by EpsilonPrime**
* Fields are now emitted in expressions completing expression handling.
    * Schemas and sources are also now implemented.
    * Removes projection as a relation detail in favor of expression for project relations.

### CODE REVIEW

File: src/substrait/textplan/parser/SubstraitPlanVisitor.cpp [hunk 1]
```diff
@@ -335,10 +351,10 @@ std::any SubstraitPlanVisitor::visitLocal_files_detail(
   for (const auto& f : ctx->file()) {
     symbolTable_->defineSymbol(
         f->getText(),
-        Location(ctx),
+        PARSER_LOCATION(ctx->parent->parent),
```
Comment (westonpace) [inline src/substrait/textplan/parser/SubstraitPlanVisitor.cpp:354]: `ctx->parent->parent` is a little confusing.  Can you add a comment here?

File: src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp [hunk 2]
```diff
@@ -435,7 +521,25 @@ std::any SubstraitPlanRelationVisitor::visitExpressionConstant(
...
+      [&](auto ref) { return (ref->name == symbolName); });
+
   ::substrait::proto::Expression expr;
+  if (currentFieldNumber != relationData->fieldReferences.end()) {
+    expr.mutable_selection()
+        ->mutable_direct_reference()
+        ->mutable_struct_field()
+        ->set_field(static_cast<int32_t>(
+            (currentFieldNumber - relationData->fieldReferences.begin()) &
+            0xffff));
```
Comment (westonpace) [inline src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp:540]: Why `& 0xFFFF`?  Should we assert that this value is less than 2 bytes?

**Review by westonpace — APPROVED** [review]
Just two minor things that I noticed.  Otherwise, I like the way this is starting to shape up

File: src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp (hunk 2 above)
Comment (EpsilonPrime) [reply src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp]: Changed to use std::numeric_limits<int32_t>::max

**Review by westonpace — APPROVED** [review]

**Pull request merged**